x-pack/plugins/maps/public/classes/sources/ems_file_source/update_source_editor.tsx

```diff
@@ -15,7 +15,7 @@ import { OnSourceChangeArgs } from '../../../connected_components/layer_panel/vi
 
 interface Props {
   layerId: string;
-  onChange: (args: OnSourceChangeArgs) => void;
+  onChange: (...args: OnSourceChangeArgs) => void;
```

Comment:
Why spread args here?

Reply:
It seemed cleaner to allow components to dispatch a single change-event with n-number of changes to the source-props (in this cases, by just using varargs), than having them call n separate dispatches. It's somewhat of a micro-optimization.